src/code/z_demo.c

```diff
@@ -1232,684 +1404,776 @@ void Cutscene_Command_Terminator(PlayState* play, CutsceneContext* csCtx, CsCmdB
+        csCtx->camAtReady = true;
+        csCtx->camAtPoints = (CutsceneCameraPoint*)script;
 
-    if ((cmdBase->startFrame < csCtx->frames) && (csCtx->frames < cmdBase->endFrame) &&
-        ((D_8015FCC0 < cmdBase->startFrame) || (D_8015FCC0 >= 0xF000))) {
-        csCtx->unk_1A = 1;
-        csCtx->subCamLookAtPoints = (CutsceneCameraPoint*)cmd;
-        if (csCtx->unk_1B != 0) {
-            D_8015FCC0 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+        if (csCtx->camEyeReady) {
+            gCamAtSplinePointsAppliedFrame = cmd->startFrame;
+
+            if (gUseCutsceneCam) {
                 Play_CameraChangeSetting(play, csCtx->subCamId, CAM_SET_CS_0);
                 Play_ChangeCameraStatus(play, sReturnToCamId, CAM_STAT_WAIT);
                 Play_ChangeCameraStatus(play, csCtx->subCamId, CAM_STAT_ACTIVE);
                 Camera_ResetAnim(Play_GetCamera(play, csCtx->subCamId));
-                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->subCamLookAtPoints,
-                                   csCtx->subCamEyePoints, GET_PLAYER(play), relativeToLink);
+                Camera_SetCSParams(Play_GetCamera(play, csCtx->subCamId), csCtx->camAtPoints, csCtx->camEyePoints,
+                                   GET_PLAYER(play), relativeToPlayer);
             }
         }
     }
 
     while (shouldContinue) {
-        if (((CutsceneCameraPoint*)cmd)->continueFlag == CS_CMD_STOP) {
-            shouldContinue = 0;
+        if (((CutsceneCameraPoint*)script)->continueFlag == CS_CMD_STOP) {
+            shouldContinue = false;
         }
-        cmd += 0x10;
-        size += 0x10;
+
+        script += sizeof(CutsceneCameraPoint);
+        size += sizeof(CutsceneCameraPoint);
     }
 
     return size;
 }
 
-// Command 0x7: ? (Related to camera positons)
-s32 Cutscene_Command_07(PlayState* play, CutsceneContext* csCtx, u8* cmd, u8 unused) {
-    CsCmdBase* cmdBase = (CsCmdBase*)cmd;
+s32 CutsceneCmd_SetCamEye(PlayState* play, CutsceneContext* csCtx, u8* script, u8 unused) {
+    CsCmdGeneric* cmd = (CsCmdGeneric*)script;
     s32 size;
-    Vec3f sp3C;
-    Vec3f sp30;
+    Vec3f at;
+    Vec3f eye;
     Camera* subCam;
-    f32 sp28;
-
-    cmd += 8;
-    size = 8;
-
-    if ((cmdBase->startFrame < csCtx->frames) && (csCtx->frames < cmdBase->endFrame) &&
-        ((D_8015FCC2 < cmdBase->startFrame) || (D_8015FCC2 >= 0xF000))) {
-        csCtx->unk_1B = 1;
-        csCtx->subCamEyePoints = (CutsceneCameraPoint*)cmd;
-        if (csCtx->unk_1A != 0) {
-            D_8015FCC2 = cmdBase->startFrame;
-            if (D_8015FCC8 != 0) {
+    f32 roll;
+
+    script += sizeof(CsCmdGeneric);
+    size = sizeof(CsCmdGeneric);
+
+    if ((csCtx->curFrame > cmd->startFrame) && (csCtx->curFrame < cmd->endFrame) &&
+        ((gCamEyeAppliedFrame < cmd->startFrame) || (gCamEyeAppliedFrame >= 0xF000))) {
+        csCtx->camEyeReady = true;
+        csCtx->camEyePoints = (CutsceneCameraPoint*)script;
+
```

Comment:
```suggestion
        csCtx->camEyeReady = true;
        
        // A single point
        csCtx->camEyePoints = (CutsceneCameraPoint*)script;

```
or something like that? since the struct member is plural, this could alleviate some confusion

Reply:
imo not necessary, its already stressed in the macros on the user interface side. and the member variable being called this is technically not wrong, its just in this context the pointer will be pointing to a single point

Reply:
the macro comment is a bit far from this line of code though
It's not a big deal though so alright